### DISCUSSION TITLE
[FEATURE REQUEST] Support new (7.8) elasticsearch index templates

### DISCUSSION THREAD

**Discussion by OrangeDog** [saltstack/salt#67663] [Feature Requests]
**Describe the solution you'd like**
Support for managing elasticsearch's composable index templates, via `elasticsearch_index_template` states.

**Describe alternatives you've considered**
Continuing to use the legacy templates, via the current `elasticsearch_index_template`.
These will eventually stop working.

**Additional context**
https://www.elastic.co/guide/en/elasticsearch/reference/7.8/index-templates.html

**Comment by DmitryKuzmenko** [comment]
@OrangeDog do you have any thoughts about the implementation. By a quick code lookup I see that we're using `elasticsearch` python library and I've found no reference to the new way using `PUT _index_template`. I'm not very good in elastic search so if you couldn't help to understand this I'll need to get a deeper research in that.

**Comment by OrangeDog** [comment]
@DmitryKuzmenko yes, the Python library also lags behind on management support.

As with the ILM (#52338) you can use the client to call arbitrary endpoints until it adds specific support.
